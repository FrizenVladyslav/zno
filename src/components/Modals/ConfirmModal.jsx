import React from 'react'
import PropTypes from 'prop-types'
import { Modal, Button, Header, Icon } from 'semantic-ui-react'

const ConfirmModal = props => {
  return (
    <Modal open={props.open} basic size="small" onClose={props.onClose}>
      <Header icon={props.icon || 'archive'} content={props.title} />
      <Modal.Content>
        <p>{props.message}</p>
      </Modal.Content>
      <Modal.Actions>
        <Button basic color="red" inverted onClick={props.onClose}>
          <Icon name="remove" /> Скасувати
        </Button>
        <Button
          color="green"
          inverted
          onClick={props.onSuccess || props.onClose}
        >
          <Icon name="checkmark" /> Підтвердити
        </Button>
      </Modal.Actions>
    </Modal>
  )
}

ConfirmModal.propTypes = {
  icon: PropTypes.string,
  message: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onSuccess: PropTypes.func,
  title: PropTypes.string.isRequired,
}

export default ConfirmModal
