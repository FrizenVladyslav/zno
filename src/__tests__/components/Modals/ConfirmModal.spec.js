import React from 'react'
import { shallow } from 'enzyme'
import ConfirmModal from 'components/Modals/ConfirmModal'

describe('confirm modal', () => {
  const initProps = {
    icon: 'icon',
    message: 'message',
    open: true,
    onClose: jest.fn(),
    onSuccess: jest.fn(),
    title: 'title',
  }
  const confirmModal = shallow(<ConfirmModal {...initProps} />)

  describe('confirm modal is open', () => {
    it('render properly', () => {
      expect(confirmModal).toMatchSnapshot()
    })
  })

  describe('when clicking close button', () => {
    beforeEach(() => {
      confirmModal
        .find('Button')
        .first()
        .simulate('click')
    })

    it('call the props.onClose', () => {
      expect(initProps.onClose).toBeCalledTimes(1)
    })
  })

  describe('when clicking success button', () => {
    beforeEach(() => {
      confirmModal
        .find('Button')
        .last()
        .simulate('click')
    })

    it('call the props.onSuccess', () => {
      expect(initProps.onSuccess).toBeCalledTimes(1)
    })
  })
})
