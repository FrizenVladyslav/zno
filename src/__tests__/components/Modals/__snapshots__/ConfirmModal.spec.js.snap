// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm modal confirm modal is open render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal
    icon="icon"
    message="message"
    onClose={[MockFunction]}
    onSuccess={[MockFunction]}
    open={true}
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "basic": true,
      "centered": true,
      "children": Array [
        <Header
          content="title"
          icon="icon"
        />,
        <ModalContent>
          <p>
            message
          </p>
        </ModalContent>,
        <ModalActions>
          <Button
            as="button"
            basic={true}
            color="red"
            inverted={true}
            onClick={[MockFunction]}
          >
            <Icon
              as="i"
              name="remove"
            />
             Скасувати
          </Button>
          <Button
            as="button"
            color="green"
            inverted={true}
            onClick={[MockFunction]}
          >
            <Icon
              as="i"
              name="checkmark"
            />
             Підтвердити
          </Button>
        </ModalActions>,
      ],
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [MockFunction],
      "open": true,
      "size": "small",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "title",
          "icon": "icon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <p>
            message
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "message",
          },
          "ref": null,
          "rendered": "message",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              as="button"
              basic={true}
              color="red"
              inverted={true}
              onClick={[MockFunction]}
            >
              <Icon
                as="i"
                name="remove"
              />
               Скасувати
            </Button>,
            <Button
              as="button"
              color="green"
              inverted={true}
              onClick={[MockFunction]}
            >
              <Icon
                as="i"
                name="checkmark"
              />
               Підтвердити
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "basic": true,
              "children": Array [
                <Icon
                  as="i"
                  name="remove"
                />,
                " Скасувати",
              ],
              "color": "red",
              "inverted": true,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "remove",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " Скасувати",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": Array [
                <Icon
                  as="i"
                  name="checkmark"
                />,
                " Підтвердити",
              ],
              "color": "green",
              "inverted": true,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " Підтвердити",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "basic": true,
        "centered": true,
        "children": Array [
          <Header
            content="title"
            icon="icon"
          />,
          <ModalContent>
            <p>
              message
            </p>
          </ModalContent>,
          <ModalActions>
            <Button
              as="button"
              basic={true}
              color="red"
              inverted={true}
              onClick={[MockFunction]}
            >
              <Icon
                as="i"
                name="remove"
              />
               Скасувати
            </Button>
            <Button
              as="button"
              color="green"
              inverted={true}
              onClick={[MockFunction]}
            >
              <Icon
                as="i"
                name="checkmark"
              />
               Підтвердити
            </Button>
          </ModalActions>,
        ],
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [MockFunction],
        "open": true,
        "size": "small",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "title",
            "icon": "icon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <p>
              message
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "message",
            },
            "ref": null,
            "rendered": "message",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                basic={true}
                color="red"
                inverted={true}
                onClick={[MockFunction]}
              >
                <Icon
                  as="i"
                  name="remove"
                />
                 Скасувати
              </Button>,
              <Button
                as="button"
                color="green"
                inverted={true}
                onClick={[MockFunction]}
              >
                <Icon
                  as="i"
                  name="checkmark"
                />
                 Підтвердити
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "basic": true,
                "children": Array [
                  <Icon
                    as="i"
                    name="remove"
                  />,
                  " Скасувати",
                ],
                "color": "red",
                "inverted": true,
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "remove",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Скасувати",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": Array [
                  <Icon
                    as="i"
                    name="checkmark"
                  />,
                  " Підтвердити",
                ],
                "color": "green",
                "inverted": true,
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "checkmark",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Підтвердити",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
